src/ipc/ipc4/handler.c

```diff
@@ -965,6 +965,37 @@ void ipc_msg_reply(struct sof_ipc_reply *reply)
 	ipc_compound_msg_done(in.primary.r.type, reply->error);
 }
 
+void ipc_msg_send_status(void)
+{
+	char *data = ipc_get()->comp_data;
+	int count = 10;
+	int ret;
+
+	/* copy mailbox data to message */
+	if (msg_reply.tx_size > 0 && msg_reply.tx_size < SOF_IPC_MSG_MAX_SIZE) {
+		ret = memcpy_s(msg_reply.tx_data, msg_reply.tx_size, data, msg_reply.tx_size);
+		if (ret) {
+			tr_err(&ipc_tr, "failed to copy memory %d", ret);
+			return;
+		}
+	}
+
+	/* send critical status to host since host is waiting for it */
+	while (count) {
+		ret = ipc_platform_send_msg(&msg_reply);
+		/* send successfully */
+		if (!ret)
+			break;
+
+		if (!--count) {
+			tr_err(&ipc_tr, "failed to send ipc status %d", ret);
+			break;
+		}
+
+		wait_delay(10000);
```

Comment:
we shouldn't use hard-coded constants like this. (10, 10000) are likely very very specific to one scheduling setting, this could be completely wrong once we start adding deep-buffer or things that take more than 1ms to complete.

At the very least an explanation of why you chose those values is required.

Reply:
sure, I will refine it. Thanks